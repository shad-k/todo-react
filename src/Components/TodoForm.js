import React, { Component } from 'react';

class TodoForm extends Component {
    constructor(props) {
        super(props);

        this.submitHandler = this.submitHandler.bind(this);
    }

    submitHandler(e) {
    	e.preventDefault();
    	const todo = this.todo.value;
    	this.props.addTodo(todo);
    	this.addForm.reset(0);
    }

    render() {
        return (
        	<form ref={(input) => this.addForm = input} onSubmit={this.submitHandler} className="add-todo-form">
	        	<input ref={(input) => this.todo = input} type="text" className="add-todo-input" />
	        	<button className="add-todo-button">Add Todo</button>
        	</form>
        );
    }
}

export default TodoForm;
