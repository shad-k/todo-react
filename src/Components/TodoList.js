import React, { Component, PropTypes } from 'react';

class TodoList extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <ul className="todo-list">
            	{this.props.todoList.map((todo, index) => {
            		return (
            			<li key={index} className="todo-item">
            				{todo}
            			</li>
            		);
            	})}
            </ul>
        );
    }
}

export default TodoList;
