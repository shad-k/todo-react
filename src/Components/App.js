import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actionCreators from '../Actions/actionCreators';
import { bindActionCreators } from 'redux';

import TodoForm from './TodoForm';
import TodoList from './TodoList';

import '../Styles/App.css';

class App extends Component {
  render() {
    return (
      <div className="container">
      	<TodoForm addTodo={this.props.addTodo} />
      	{this.props.todos.length !== 0 &&
      		<TodoList todoList={this.props.todos} />
      	}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
	return {
		todos: state.todos
	};
}

const mapDispatchToProps = (dispatch) => {
	return bindActionCreators(actionCreators, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
