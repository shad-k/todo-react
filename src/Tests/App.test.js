//https://academy.plot.ly/react/6-testing/
import React from 'react';
import renderer from 'react-test-renderer';
import { App } from '../Components/App';
import TodoForm from '../Components/TodoForm';
import TodoList from '../Components/TodoList';

describe("components", () => {
	describe("<App />", () => {
		it('renders without crashing', () => {
		  const tree = renderer.create(<App />).toJSON();
		  expect(tree).toMatchSnapshot();
		});
	});

	describe("<TodoForm />", () => {
		it('renders without crashing', () => {
		  const tree = renderer.create(<TodoForm />).toJSON();
		  expect(tree).toMatchSnapshot();
		});
	});

	describe("<TodoList />", () => {
		it('renders without crashing', () => {
		  const tree = renderer.create(<TodoList todoList={[]}/>).toJSON();
		  expect(tree).toMatchSnapshot();
		});
	});
});
