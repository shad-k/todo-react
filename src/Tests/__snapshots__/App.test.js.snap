// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <App /> renders without crashing 1`] = `
<div
  className="container"
>
  <form
    className="add-todo-form"
    onSubmit={[Function]}
  >
    <input
      className="add-todo-input"
      type="text"
    />
    <button
      className="add-todo-button"
    >
      Add Todo
    </button>
  </form>
</div>
`;

exports[`components <TodoForm /> renders without crashing 1`] = `
<form
  className="add-todo-form"
  onSubmit={[Function]}
>
  <input
    className="add-todo-input"
    type="text"
  />
  <button
    className="add-todo-button"
  >
    Add Todo
  </button>
</form>
`;

exports[`components <TodoList /> renders without crashing 1`] = `
<ul
  className="todo-list"
/>
`;
